pkg/bus/bus.go

```diff
@@ -155,13 +162,68 @@ func (b *InProcBus) Dispatch(msg Msg) error {
 	return err.(error)
 }
 
+// PublishCtx function publish a message to the bus listener.
+func (b *InProcBus) PublishCtx(ctx context.Context, msg Msg) error {
+	var msgName = reflect.TypeOf(msg).Elem().Name()
+	withCtx := true
+	var listeners = b.listenersWithCtx[msgName]
+	if listeners == nil {
+		withCtx = false
+		listeners = b.listeners[msgName]
+		if listeners == nil {
+			return ErrListenerNotFound
```

Comment:
Comparing the old code, this case shouldn't result in an error - just return nil. Meaning that if something is published and there are no listeners, that's completely valid.